operator/builtin/transformer/add/add.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package add
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/antonmedv/expr"
+	"github.com/antonmedv/expr/vm"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+)
+
+func init() {
+	operator.Register("add", func() operator.Builder { return NewAddOperatorConfig("") })
+}
+
+// NewAddOperatorConfig creates a new add operator config with default values
+func NewAddOperatorConfig(operatorID string) *AddOperatorConfig {
+	return &AddOperatorConfig{
+		TransformerConfig: helper.NewTransformerConfig(operatorID, "add"),
+	}
+}
+
+// AddOperatorConfig is the configuration of an add operator
+type AddOperatorConfig struct {
+	helper.TransformerConfig `mapstructure:",squash" yaml:",inline"`
+	Field                    entry.Field `mapstructure:"field" json:"field" yaml:"field"`
+	Value                    interface{} `mapstructure:"value,omitempty" json:"value,omitempty" yaml:"value,omitempty"`
+}
+
+// Build will build an add operator from the supplied configuration
+func (c AddOperatorConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	transformerOperator, err := c.TransformerConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	addOperator := &AddOperator{
+		TransformerOperator: transformerOperator,
+		Field:               c.Field,
+	}
+	strVal, ok := c.Value.(string)
+	if ok && strings.HasPrefix(strVal, "EXPR(") && strings.HasSuffix(strVal, ")") {
+		exprStr := strings.TrimPrefix(strVal, "EXPR(")
+		exprStr = strings.TrimSuffix(exprStr, ")")
+
+		compiled, err := expr.Compile(exprStr, expr.AllowUndefinedVariables())
+		if err != nil {
+			return nil, fmt.Errorf("failed to compile expression '%s': %w", c.IfExpr, err)
+		}
+
+		addOperator.program = compiled
+		return []operator.Operator{addOperator}, nil
+	}
+	addOperator.Value = c.Value
+	return []operator.Operator{addOperator}, nil
```

Comment:
Let's extract the expression check and put the simple case first so it can return right away:

```
func isExpr(str string) bool {
	strings.HasPrefix(strVal, "EXPR(") && strings.HasSuffix(strVal, ")")
}

	...
	strVal, ok := c.Value.(string)
	if !ok || !isExpr(strVal) {
		addOperator.Value = c.Value
		return []operator.Operator{addOperator}, nil
	}

	exprStr := strings.TrimPrefix(strVal, "EXPR(")
	exprStr = strings.TrimSuffix(exprStr, ")")

	compiled, err := expr.Compile(exprStr, expr.AllowUndefinedVariables())
	if err != nil {
		return nil, fmt.Errorf("failed to compile expression '%s': %w", c.IfExpr, err)
	}

	addOperator.program = compiled
	return []operator.Operator{addOperator}, nil
```